### PR TITLE
Refactor to user invite creation

### PR DESCRIPTION
**Pull request by nahidalgo**
Added fiscalCode, name and surname to invite creation.
Extra controls to check if the user already exists.

### CODE REVIEW

File: user/create-invite.go [hunk 1]
```diff
@@ -24,32 +27,58 @@ func CreateInviteFx(w http.ResponseWriter, r *http.Request) (string, interface{}
 		return `{"success": false}`, `{"success": false}`, nil
 	}
 
-	if inviteUid, ok := CreateInvite(createInviteRequest.Email, createInviteRequest.Role, r.Header.Get("Origin"), creatorUid); ok {
+	if inviteUid, err := CreateInvite(createInviteRequest.Name, createInviteRequest.Surname, createInviteRequest.FiscalCode, createInviteRequest.Email, createInviteRequest.Role, r.Header.Get("Origin"), creatorUid); err != nil {
```
Comment (Yousef-Hammar) [inline user/create-invite.go]: Pass entire `CreateInviteRequest` struct as input to `CreateInvite` function

File: user/consume-invite.go [hunk 2]
```diff
@@ -56,12 +56,15 @@ func ConsumeInvite(inviteUid, password, origin string) (bool, error) {
...
+		Uid:        userRecord.UID,
+		AuthId:     userRecord.UID,
+		Role:       invite.Role,
+		FiscalCode: invite.FiscalCode,
+		Name:       invite.Name,
+		Surname:    invite.Surname,
 	}
-	
+
 	lib.SetFirestore(usersCollectionName, user.Uid, user)
```
Comment (diogo-wopta) [inline user/consume-invite.go]: Shouldn't we use the new function SetFirestoreErr and handle eventual errors?

File: user/user.go [hunk 3]
```diff
@@ -169,3 +169,14 @@ func SetUserIntoPolicyContractor(policy *models.Policy, origin string) {
 		lib.CheckError(err)
 	}
 }
+
+func GetAuthUserByMail(mail string) (models.User, error) {
+	var user models.User
+
+	authId, err := lib.GetAuthUserIdByEmail(mail)
```
Comment (diogo-wopta) [inline user/user.go:176]: Even though both have the same value (as seen on consume-invite.go lines 60, 61), the function GetAuthUserIdByEmail returns the UID, not the authId.

Comment (nahidalgo) [reply user/user.go:176]: The user in `GetAuthUserIdByEmail` is not a `User` from models, instead, it is a Firebase User, which the UID is the same as our `models.User.Uid` and `models.User.AuthId`.

**Review by diogo-wopta — APPROVED** [review]

**Review by Yousef-Hammar — APPROVED** [review]

**Pull request merged**